internal/entities/properties/properties.go

```diff
@@ -20,17 +20,22 @@ import (
 	"fmt"
 
 	"github.com/puzpuzpuz/xsync/v3"
+	"google.golang.org/protobuf/types/known/structpb"
 )
 
 // Property is a struct that holds a value. It's just a wrapper around an interface{}
 // with typed getters and handling of an empty receiver
 type Property struct {
-	value any
+	value *structpb.Value
```

Comment:
There are several reasons why I chose `structpb.Value` over just `any`:
 - `NewValue` contains pretty nice code that can e.g. serialize a `[]byte` slice base64 encoded
 - Returning the value as `any` actually does checks, for example for over/underflow for numerical values
 - there is another draft PR that proposes to use `structpb.Struct` in the general entity protobuf message which would make the value easily usable should we need to

Reply:
I dig it!